bitcoin/src/address.rs

```diff
@@ -589,7 +678,7 @@ impl Address {
     /// these days.
     #[inline]
     pub fn p2sh(script: &script::Script, network: Network) -> Result<Address, Error> {
-        Ok(Address { network, payload: Payload::p2sh(script)? })
+        Ok(Address { network, payload: Payload::p2sh(script)?, _validation: PhantomData })
```

Comment:
I think we should have an internal `new()` method to avoid adding `, _validation: PhantomData` all over the place.